### ISSUE TITLE
Mathesar should not depend on custom types being installed.

### ISSUE THREAD

**Issue by kgodey** [mathesar-foundation/mathesar#403]
## Problem

We currently assume that custom types have been installed in the database in various places in the Mathesar codebase. This is a faulty assumption - Mathesar should work well with whatever types are available. Custom types should only provide additional functionality, not be a requirement.

## Proposed solution

We need to ensure that there is no assumption that custom types are installed anywhere in the codebase. We should detect the types that are installed and only use those for:
- type inference during file import
- the API listing types available in the database
- the column API listing types that a  given column can be altered to
- any SQLAlchemy engines that are created in the codebase 

We should also check the codebase for other places that types are used to make sure there's no other places that need to be updated.

## Additional context

- https://github.com/centerofci/mathesar/discussions/355 has more context on current behavior.
- The conversation on #532 may be useful.
- Discussion on #347 inspired this issue:
   - There are a lot of comments on that PR, relevant ones: [one](https://github.com/centerofci/mathesar/pull/347#issuecomment-876871504), [two](https://github.com/centerofci/mathesar/pull/347#issuecomment-877245010), [three](https://github.com/centerofci/mathesar/pull/347#issuecomment-877373807), [four](https://github.com/centerofci/mathesar/pull/347#issuecomment-877754289), [five](https://github.com/centerofci/mathesar/pull/347#issuecomment-878292814).
- #269 may be useful to look at.  
- [Mathesar's Product Principles](https://wiki.mathesar.org/product/principles)
- We'll need to make sure to thoroughly document the different options with custom types when we create installation documentation for Mathesar. That doesn't seem within scope for this issue, since we have no installation documentation yet.

**Comment by mathemancer** [comment]
It may be possible to solve _all_ of these by altering the `get_supported_alter_column_types` method in the `db.types.alteration` module.

**Comment by mathemancer** [comment]
@pavish Not quite. At least inference will break if we don't have all types installed. Potentially other things as well.

**Comment by kgodey** [comment]
Closing this since it's out of date, we'll track this in our roadmap soon.
